src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -326,14 +326,15 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
     const SchedulingKey &scheduling_key, const rpc::Address *raylet_address) {
   auto &scheduling_key_entry = scheduling_key_entries_[scheduling_key];
 
-  if (scheduling_key_entry.pending_lease_requests.size() ==
-      max_pending_lease_requests_per_scheduling_category_) {
+  uint64_t max_pending_lease_requests =
+      std::max(get_max_pending_lease_requests_per_scheduling_category_(),
```

Comment:
nit: should assert that the function is valid via https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool in the constructor